test/system/211-dbus.bats

```diff
@@ -58,22 +58,372 @@ teardown() {
+                         --method org.freedesktop.DBus.Peer.Ping)"
+
+  create_distro_container fedora 34 fedora-toolbox-34
+
+  run --keep-empty-lines --separate-stderr "$TOOLBOX" run \
+    --distro fedora \
+    --release 34 \
+    gdbus call \
+      --session \
+      --dest org.freedesktop.DBus \
+      --object-path /org/freedesktop/DBus \
+      --method org.freedesktop.DBus.Peer.Ping
+
+  assert_success
+  assert_line --index 0 "$expected_response"
+
+  if check_bats_version 1.10.0; then
+    assert [ ${#lines[@]} -eq 1 ]
+  else
+    assert [ ${#lines[@]} -eq 2 ]
+  fi
+
+  # shellcheck disable=SC2154
+  assert [ ${#stderr_lines[@]} -eq 0 ]
+}
+
+@test "dbus: session bus inside RHEL 8.7" {
+  local expected_response
+  expected_response="$(gdbus call \
+                         --session \
+                         --dest org.freedesktop.DBus \
+                         --object-path /org/freedesktop/DBus \
+                         --method org.freedesktop.DBus.Peer.Ping)"
+
+  create_distro_container rhel 8.7 rhel-toolbox-8.7
+
+  run --keep-empty-lines --separate-stderr "$TOOLBOX" run --distro rhel --release 8.7 gdbus call \
+                                                                                        --session \
+                                                                                        --dest org.freedesktop.DBus \
+                                                                                        --object-path /org/freedesktop/DBus \
+                                                                                        --method org.freedesktop.DBus.Peer.Ping
+
+  assert_success
+  assert_line --index 0 "$expected_response"
+
+  if check_bats_version 1.10.0; then
+    assert [ ${#lines[@]} -eq 1 ]
+  else
+    assert [ ${#lines[@]} -eq 2 ]
+  fi
+
+  # shellcheck disable=SC2154
+  assert [ ${#stderr_lines[@]} -eq 0 ]
+}
+
+@test "dbus: session bus inside Ubuntu 16.04" {
+  local expected_response
+  expected_response="$(busctl call \
+                        --user org.freedesktop.DBus \
+                        /org/freedesktop/DBus \
+                        org.freedesktop.DBus.Peer Ping)"
+
+  create_distro_container ubuntu 16.04 ubuntu-toolbox-16.04
+
+  run --keep-empty-lines --separate-stderr "$TOOLBOX" run --distro ubuntu --release 16.04 busctl call \
+                                                                                            --user \
+                                                                                            org.freedesktop.DBus \
+                                                                                            /org/freedesktop/DBus \
+                                                                                            org.freedesktop.DBus.Peer \
+                                                                                            Ping
+
+  assert_success
+  assert_equal "${lines[*]}" "$expected_response"
```

Comment:
Umm... why not: `assert [ ${#lines[@]} -eq 0 ]` ?

That's easier to read, and we use it in many other places.  eg., the assertion about `stderr_lines` below.

Reply:
Done.